controllers/conditions.go

```diff
@@ -17,60 +17,135 @@ limitations under the License.
+		Reason:  reason,
+		Message: message,
+	})
 }
 
-func GetConditionResourcePresent(status corev1.ConditionStatus, reason, msg string) conditionsv1.Condition {
-	return GetCondition(odfv1alpha1.ConditionResourcePresent, status, reason, msg)
+func SetReconcileAccomplishCondition(conditions *[]conditionsv1.Condition,
+	reason string, message string) {
+
+	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionAvailable,
+		Status:  corev1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionProgressing,
+		Status:  corev1.ConditionFalse,
+		Reason:  reason,
+		Message: message,
+	})
 }
 
-func (r *StorageSystemReconciler) setConditionResourcePresent(instance *odfv1alpha1.StorageSystem, logger logr.Logger) (bool, error) {
-
-	var requeue bool
-	var err error
-
-	if instance.Spec.Kind == VendorStorageCluster() {
-		logger.Info("get storageCluster")
-		storageCluster := &ocsv1.StorageCluster{}
-		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.Name, Namespace: instance.Spec.Namespace}, storageCluster)
-	} else if instance.Spec.Kind == VendorFlashSystemCluster() {
-		logger.Info("get flashSystemCluster")
-		flashSystemCluster := &ibmv1alpha1.FlashSystemCluster{}
-		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.Name, Namespace: instance.Spec.Namespace}, flashSystemCluster)
-	}
-
-	if err == nil {
-		logger.Info("set condition", string(odfv1alpha1.ConditionResourcePresent), corev1.ConditionTrue)
-		conditionsv1.SetStatusCondition(&instance.Status.Conditions, GetConditionResourcePresent(corev1.ConditionTrue, "Found", ""))
-		requeue = false
-	} else if errors.IsNotFound(err) {
-		logger.Error(err, "set condition", string(odfv1alpha1.ConditionResourcePresent), corev1.ConditionFalse)
-		conditionsv1.SetStatusCondition(&instance.Status.Conditions, GetConditionResourcePresent(corev1.ConditionFalse, "NotFound", err.Error()))
-		requeue = true
-	} else {
-		logger.Error(err, "set condition", string(odfv1alpha1.ConditionResourcePresent), corev1.ConditionUnknown)
-		conditionsv1.SetStatusCondition(&instance.Status.Conditions, GetConditionResourcePresent(corev1.ConditionUnknown, "", err.Error()))
-		requeue = true
-	}
-
-	return requeue, err
+func SetDeleteionInProgressCondition(conditions *[]conditionsv1.Condition,
+	reason string, message string) {
+
+	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionAvailable,
+		Status:  corev1.ConditionFalse,
+		Reason:  reason,
+		Message: message,
+	})
+	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionProgressing,
+		Status:  corev1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func SetStorageSystemInValidCondition(conditions *[]conditionsv1.Condition,
```

Comment:
Treat this function like you do the other two. Only focus on one condition and take the status as a parameter. In particular, you should be setting invalid=false when we have validated the spec.

Reply:
In that case, According to that, we do need to change progress also.